style-sync/src/main/scala/commercialtools/stylesync/StyleSync.scala

```diff
@@ -0,0 +1,146 @@
+              dfpFormat = format,
+              styleName = styleName,
+              dfpStyle = style,
+              localPath = styleDirectory.getAbsolutePath,
+              testFile)
+    }
+  }
+
+  case class DiffLine(line: String, number: Int)
+  case class DiffResult(diffLines: List[DiffLine], sourceALines: List[String], sourceBLines: List[String])
+
+  def summarise(formats: List[CreativeTemplate], styles: List[NativeStyle], guStyles: List[GuStyle]) = {
+
+    val divider: String = "------------------------------------------------------------------"
+
+    def printSummary(message: String, elements: List[String]): Unit = {
+      println(divider)
+      println(message)
+      elements foreach { element => println(s"\t$element")}
+    }
+
+    def compareLines(sourceA: List[String], sourceB: List[String]): DiffResult = {
+      val diffLines = sourceA.filterNot(a => sourceB.contains(a)).zipWithIndex map DiffLine.tupled
+      DiffResult(diffLines, sourceA, sourceB)
+    }
+
+    def summariseDiff(diffResult: Option[DiffResult]) = {
+      diffResult match {
+        case None => s"missing"
+        case Some(result) if result.diffLines.isEmpty => "matches"
+        case Some(result) => "differs (%d/%d)".format(result.diffLines.size, result.sourceALines.size)
+      }
+    }
+
+    val formatsMissingInDFP: List[String] = guStyles filter (_.dfpFormat.isEmpty) map (_.formatName)
+    printSummary("1. Formats found locally, not in DFP:", formatsMissingInDFP)
+
+    val stylesMissingInDFP: List[String] = guStyles filter (_.dfpStyle.isEmpty) map (_.styleName)
+    printSummary("2. Styles found locally, not in DFP:", stylesMissingInDFP)
+
+    val formatsMissingInConfigs: List[String] = formats filter { format =>
+      !guStyles.exists(_.formatName == format.getName)
+    } map (_.getName)
+    printSummary("3. Formats in DFP, not found locally:", formatsMissingInConfigs)
+
+    val stylesMissingInConfigs: List[String] = styles filter { style =>
+      !guStyles.exists(_.styleName == style.getName)
+    } map (_.getName)
+    printSummary("4. Styles in DFP, not found locally:", stylesMissingInConfigs)
+
+    println("5. Comparison of HTML/CSS locally and in DFP:")
+    for (guStyle <- guStyles) {
+
+      val styleString: String = s"Style: ${guStyle.styleName}"
+
+      val cssDiffLines: Option[DiffResult] = guStyle.dfpStyle map (dfpStyle => compareLines(guStyle.localCssLines, dfpStyle.getCssSnippet.split("\n").toList))
+      val cssResult = s"CSS: ${summariseDiff(cssDiffLines)}"
+
+      val htmlDiffLines: Option[DiffResult] = guStyle.dfpStyle map (dfpStyle => compareLines(guStyle.localHtmlLines, dfpStyle.getHtmlSnippet.split("\n").toList))
+      val htmlResult = s"HTML: ${summariseDiff(htmlDiffLines)}"
+
+      val testFileResult = s"Test File: ${if (guStyle.testFile.isDefined) "found" else "missing"}"
+
+      println("\t%1$-50s %2$-25s %3$-25s %4$-20s".format(styleString, htmlResult, cssResult, testFileResult))
```

Comment:
What is this?  May be worth assigning string to named val for clarity.

Reply:
Yeah good point. It is basically the formatting string for printing a well-spaced row for each style in the summary.